### PR TITLE
Add --edit CLI mode

### LINKED ISSUE #18: Editing Tools
Explore alternative approaches to editing tool calling. The first goal is that Ailly will be able to apply _and undo_ edits to larger source code files; the second goal is for Ailly to execute those files and iterate on the execution output.

* https://python.langchain.com/docs/modules/model_io/chat/function_calling
* https://docs.anthropic.com/claude/docs/functions-external-tools
* https://medium.com/@zeek.granston/function-calling-with-anthropic-claude-and-amazon-bedrock-c6eda7358b0f

### PR DESCRIPTION
**Pull request by DavidSouther**
--edit --lines has the CLI attempt to replace --lines in the single target file with the response from the LLM call.

Closes #18

### CODE REVIEW

File: cli/fs.js [hunk 1]
```diff
@@ -34,7 +34,34 @@ export async function loadFs(args) {
...
+    const start = Number(line[0]) - 1;
+    const end = Number(line[1]) - 1;
+    switch (true) {
+      case hasStart && hasEnd:
+        edit = { start, end, file: '' };
+        break;
+      case hasStart:
+        edit = { start, end: start + 1, file: '' };
+        break;
+      case hasEnd:
```
Comment (cpyle0819) [inline cli/fs.js:59]: What about `hasStart === false && hasEnd === false`?

File: cli/fs.js [hunk 2]
```diff
@@ -54,7 +81,13 @@ export async function loadFs(args) {
   }
 
   if (hasPrompt) {
+    if (edit && content.length == 1) {
+      edit.file = content[0];
+      content = [];
+    }
     const noContext = args.values.context == "none";
+    const folder = args.values.context == 'folder' ?
+      Object.values(context).find(c => dirname(c.path) == root)?.context.folder : edit ? [edit.file] : undefined;
```
Comment (cpyle0819) [inline cli/fs.js:90]: Ok, so no start/end means `context.folder === undefined`

File: cli/fs.js (hunk 1 above)
Comment (DavidSouther) [reply cli/fs.js:59]: Initialized to `undefined`, stays `undefined`. (This does get refactored in the next PR)

File: cli/fs.js (hunk 2 above)
Comment (DavidSouther) [reply cli/fs.js:90]: if context is folder, use the whole folder; otherwise, if editing, folder is the one file; otherwise, no folder.

**Review by cpyle0819 — APPROVED** [review]

**Pull request merged**